x-pack/plugin/ent-search/src/test/java/org/elasticsearch/xpack/application/EnterpriseSearchModuleTestUtils.java

```diff
@@ -87,18 +86,77 @@ public static Map<String, Object> randomSearchApplicationQueryParams() {
     }
 
     public static QueryRuleCriteria randomQueryRuleCriteria() {
-        // We intentionally don't allow ALWAYS criteria in this method, since we want to test parsing metadata and values
-        QueryRuleCriteriaType type = randomFrom(Arrays.stream(QueryRuleCriteriaType.values()).filter(t -> t != ALWAYS).toList());
-        return new QueryRuleCriteria(type, randomAlphaOfLengthBetween(1, 10), randomList(1, 5, () -> randomAlphaOfLengthBetween(1, 10)));
+        QueryRuleCriteriaType criteriaType = randomFrom(QueryRuleCriteriaType.values());
+        String metadata = randomAlphaOfLength(10);
+        List<Object> values = new ArrayList<>();
+        int numValues = randomIntBetween(1, 3);
+
+        if (criteriaType.isNumericComparison()) {
```

Comment:
If you remove `isNumericComparison` you can directly check the types here rather than creating the methods to create numeric/nonNumeric criteria